packages/abi-coder/src/coders/byte.ts

```diff
@@ -43,14 +43,14 @@ export class ByteCoder extends Coder<number[], Uint8Array> {
   }
 
   #getPaddedData(value: number[]): Uint8Array {
-    const data: Uint8Array[] = [arrayify(value)];
+    const data: Uint8Array[] = [Uint8Array.from(value)];
 
     const paddingLength = (WORD_SIZE - (value.length % WORD_SIZE)) % WORD_SIZE;
     if (paddingLength) {
       data.push(new Uint8Array(paddingLength));
     }
 
-    return concat(data);
+    return concatBytes(data);
```

Comment:
Usage of new utility function `concatBytes`, where the ethers v6 `concat` is no longer a viable solution as it will only return a hex string.